### ISSUE TITLE
Last working version for 1080p live streaming is 0.3.5 ?!

### ISSUE THREAD

**Issue by KingKebab** [mangui/flashls#425]
Hello,

I was looking for an upgrade about the HLS Provider extension for OSMF clone players, all is working fine with the last FlasHLS  0.4.4.4. But 1080p HLS streams have some randomly stops. Only 1080p quality and over have this problem. I have the same issue with single bitrate and multi-bitrate (Adaptive Streaming).

I tested many releases, and here is my conclusion :
I can play the same stream without any stop with FlasHLS 0.3.5, but FlasHLS 0.4.0 and all later releases are not 1080p friendly. I can't reproduce with pre-encoded HLS video streams, only HLS live streams.

I tested other HLS video players using Flash into 1080p : MediaElement and Kutu Player are working properly, BitDash have the same problem as latests FlasHLS releases.

I produce my HLS live stream with Vodobox Live Server since 2 years. It's working with all Apple devices and operating system since iOS 3.x. Do I have something special (some parameters) to define with FlasHLS 0.4.4.4 to support 1080p live streaming again ?

VLC Media Player, Quicktime (iOS+Mac OS) and Windows Edge HTML5 video tag support my live stream without randomly stops too.

**Comment by mangui** [comment]
Could u share a stream so that I can look into it ?

**Comment by KingKebab** [comment]
Hello Guillaume,

I can't share this live stream, it's under restricted access. But, you should be able to reproduce my live stream with Vodobox Live Server. Software is available as unlimited trial for testing :
http://www.vodobox.com/index.html?content=download_live_server_en.html

To create an HLS 1080p test live stream, you have to :
. click on the dowload sample to get a video source (Big Buck Bunny)
. select H264_1080p as video output quality in Transcode panel
. then click Start button and your live Stream is available at http://%YOUR_LOCAL_IP%/stream.m3u8
. that's all

Then you'll be able to play the live stream simultaneously with your online players. 

FlasHLS v0.3.5 :
http://flashls.org/flashls-0.3.5/examples/osmf/GrindPlayer.html?src=http%3A%2F%2F192.168.1.3%2Fstream.m3u8 

FlasHLS v0.4.4.4 :
http://flashls.org/flashls-0.4.4.4/examples/osmf/GrindPlayer.html?src=http%3A%2F%2F192.168.1.3%2Fstream.m3u8

At a moment, 0.4.x releases will stop playing while 0.3.5 release will continue to read the live stream.

**Comment by mangui** [comment]
I don't use MS Windows (Vodobox needs it) and I just have bw to troubleshoot playlists
if you want to get some help plz share a test stream or a snapshot of your live stream.

**Comment by KingKebab** [comment]
Hello Guillaume,

I'm sorry, I made a mistake in my first report. The latest FlasHLS version optimized for 1080p live streams is 0.4.0.5 and not 0.3.5.

I made a video desktop capture where you can see 0.4.0.6 and 0.4.4.4 releases stopping on the same 1080p HLS live stream, while 0.4.0.5 and VLC Media Player continue to play. In my video record you can see also, 720p is optimized with all FlashHLS version :
https://www.youtube.com/watch?v=YxT99Za9A_s

I just tested the new 0.4.4.5 release with Chromless, Strobe and Grind players. The problem hasn't gone away.

Unfortunately I can't share the 1080p live stream. Because I'm not the service owner and access to these video streams is allowed for its subscribers only. Share my test account with anyone is a professional fault.

This customer ask me to upgrade its stream into Full HD. So, at the same time I was looking to upgrade its video player for any better performance. Don't worry, I can use the FlasHLS 0.4.0.5 release instead of the newest release. Now, it's playing since 16 hours without stops.

It's really strong what you have achieved in this project.
Thank you so much.

**Comment by mangui** [comment]
Your YouTube video seems private. 
It is weird to have regression between 0.4.0.5 and 0.4.0.6. Maybe the changes around the TS demuxer ... you could maybe PM an URL to me. You can find my email on github and git commit.

**Comment by KingKebab** [comment]
Oups ! You're right, I forgot to share the video. It's fixed.

My home connection is too slow to share a 1080p live Stream. But I'll try to put a Vodobox Live Server on a Windows Azure VPS next week. I have to ask a friend to get remote access.

Have nice WE :)

**Comment by mangui** [comment]
could you share debug/debug2 logs ?
refer to https://github.com/mangui/flashls#configuration

**Comment by KingKebab** [comment]
Hello Guillaume,

You can find in attachment a screenshot and a firebug log file made with FlasHLS 0.4.4.5. In this snapshot, the live streaming stops at 79.ts and then asks for the m3u8 playlist 4 times. Sometimes it can ask the m3u8 file, 6 or 8 times after. And finally the live stream stops definitely.

![firefox_screenshot](https://cloud.githubusercontent.com/assets/15297264/10880476/bb95a2a2-815b-11e5-8d43-366b6451cec1.png)
Logfile download : http://www.filedropper.com/firebug-console-log

Otherwise, you can test my BBBunny live stream via this URL :
http://mercredi.cloudapp.net/stream.m3u8

Or in your Chromeless player :
http://flashls.org/flashls-0.4.4.5/examples/chromeless/?src=http%3A%2F%2Fmercredi.cloudapp.net%2Fstream.m3u8

VODOBOX Live Server is looping on the same video source file and provides a 1080p live stream at 4500 kbps from a Windows Azure cloud server. This server isn't mine, but I can let it play during few times to let you test.

Keep me informed when I can stop the server. 
I should be able to restart it again, if required later.

**Comment by KingKebab** [comment]
I stopped the VPN.
If required, ask me and I'll restart it.

**Comment by mangui** [comment]
@KingKebab could you restart it tomorrow during working hour, CET timezone ?

**Comment by KingKebab** [comment]
Hello Guillaume,

BBBunny 1080p live stream is awake :
http://flashls.org/flashls-0.4.4.5/examples/chromeless/?src=http%3A%2F%2Fmercredi.cloudapp.net%2Fstream.m3u8

You can check it now, I will let it run all the day.

**Comment by mangui** [comment]
the stream seems quite unstable. 
i was able to make it work once, but it ended in I/O timeout while refreshing the live manifest.
now i can't even get the manifest http://mercredi.cloudapp.net/stream.m3u8

**Comment by KingKebab** [comment]
I'm sorry and I confirm !

The VM has decided to operate a Windows Update by itself.
CPU load was 100% and now it's restarting.

Live stream is back again and CPU charge is back to a normal charge.
Windows Update and TrutedInstaller.exe service are now disabled.
It shouldn't happen again.

**Comment by KingKebab** [comment]
I just made a big mistake in Azure console.
I deleted the VM by error !

I can reinstall.
I inform when all is back for clean.
It's not my day today.

**Comment by KingKebab** [comment]
Done !

I reinstalled on a Windows 2012 R2 instead of 2008 Server.

BBBunny 1080p live stream is really awake :
http://flashls.org/flashls-0.4.4.5/examples/chromeless/?src=http%3A%2F%2Fmercredi.cloudapp.net%2Fstream.m3u8

I let play for now and don't touch anything.

**Comment by mangui** [comment]
it always fail for the same reason.
after some time (several minutes, maybe when the stream is relooping ?),
i got multiple (3) 404 error on the m3u8
`stream.m3u8:1 GET http://mercredi.cloudapp.net/stream.m3u8 404 (Not Found)`

you can try to increase this value
https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/HLSSettings.as#L253
using config params https://github.com/mangui/flashls#configuration

**Comment by KingKebab** [comment]
Yes that's what I also found by monitoring the network traffic in Firebug. Multiple M3U8 requests then it stops. It happens randomly. Sometimes before 1 minutes playback, sometimes after 5 minutes.

It only happens with 1080p streams, never with 720p streams or lower quality. It only happens with FlasHLS 0.4.0.6 and later releases. 0.4.0.5 and earlier releases works great.

In my video player, I already used hls_manifestloadmaxretry : -1 to enable M3U8 infinite retry. Unfortunately it doesn't solve.

I stopped the Azure server for now. Vodobox support recommends to use Viblast player instead. But I prefer use an older FlasHLS release. It's not a problem.

Thank you for tested my live stream today.
